### PR TITLE
updated http GETs methods to utilize new library

### PR DESCRIPTION
**Pull request by RejectKid**
As stated in our discussions.  so far just the base class is converted over, since the child class inheriting from requester is using the tounery API (which Ive seen talks is broken currently) I've left them pointing to the old methods. Ideally I'd like to clean those up as well and just have the new API calls

### CODE REVIEW

**Comment by BenFradet** [comment]
The tournament API is not broken, if you have a tournament API key it should run fine.
However, some tests are broken regarding the league endpoint due to the new season.

**Comment by RejectKid** [comment]
I briefly looked around, on how to get the tourney API key and failed.  where is it that I can obtain this?

**Comment by BenFradet** [comment]
On https://developer.riotgames.com/, you have to register a new project and specify in the options that you'll need access to the tournament api key.

**Comment by BenFradet** [comment]
By the way, the broken tests are now fixed so if you merge/rebase with an updated version of develop all tests should run fine.

**Comment by RejectKid** [comment]
ah perfect, Ive requested the tourney api and once received ill work on the rest of the transferring over

**Comment by BenFradet** [comment]
Great!

**Comment by RejectKid** [comment]
so looks like this is good to go.  I also changed the PUT's async test as it didnt match the non-async one to check to see if pickType and Map changed.  All tests pass as well

**Comment by BenFradet** [comment]
Great, I will have to check that out next week

**Comment by BenFradet** [comment]
Will review tomorrow, sorry for the delay.

File: RiotSharp/Requester.cs [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 using System.Net;
+using System.Net.Http;
 using System.Threading.Tasks;
 
 namespace RiotSharp
 {
     class Requester
     {
-        protected string rootDomain;
+        protected string RootDomain;
```
Comment (BenFradet) [inline RiotSharp/Requester.cs]: why `RootDomain` instead of `rootDomain`?

File: RiotSharp/Requester.cs [hunk 2]
```diff
@@ -19,72 +20,112 @@ internal Requester(string apiKey = "")
...
+        }
+
+        protected string GetResult(HttpRequestMessage request)
+        {
+            var result = string.Empty;
+            try
             {
-                request = (HttpWebRequest)WebRequest.Create(string.Format("{0}://{1}{2}?api_key={3}"
-                    , scheme, rootDomain, relativeUrl, ApiKey));
+                using (var client = new HttpClient())
```
Comment (BenFradet) [inline RiotSharp/Requester.cs]: Do we have to create a new `HttpClient` for each request?
Can't we create one as a class variable?

File: RiotSharp/TournamentRiotApi.cs [hunk 3]
```diff
@@ -81,6 +81,9 @@ internal static TournamentRiotApi GetInstance()
         public TournamentProvider CreateProvider(Region region, string url)
         {
             var body = new Dictionary<string, object> {{"url", url}, {"region", region.ToString().ToUpper()}};
+
```
Comment (BenFradet) [inline RiotSharp/TournamentRiotApi.cs]: Why the empty lines?

File: RiotSharpTest/App.config [hunk 4]
```diff
@@ -1,8 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8" ?>
 <configuration>
     <appSettings>
-        <add key="ApiKey" value="API_KEY"/>
-        <add key="TournamentApiKey" value="TOURNAMENT_API_KEY"/>
+        <add key="ApiKey" value="fb4dbd1e-7da8-4585-9efa-c98859d91ca3"/>
```
Comment (BenFradet) [inline RiotSharpTest/App.config]: oops, you've commited your api keys.

**Comment by BenFradet** [comment]
LGTM except for a few minor comments.
I'll try that out tonight.

File: riot.txt [hunk 5]
```diff
@@ -1 +1,2 @@
 02da31c6-6e39-443c-822f-2c6ef01f1fa0
+18d930c0-b69f-4ae6-8e0c-00074e4048a7
```
Comment (BenFradet) [inline riot.txt]: I don't think this is necessary since I'm the sole maintainer of the project.

File: RiotSharp/Requester.cs (hunk 1 above)
Comment (RejectKid) [inline RiotSharp/Requester.cs]: its just a style thing, from the default Resharper settings.  Would you like me to change it back?

File: RiotSharpTest/App.config (hunk 4 above)
Comment (RejectKid) [inline RiotSharpTest/App.config]: oops ill fix that

File: RiotSharp/Requester.cs (hunk 2 above)
Comment (RejectKid) [inline RiotSharp/Requester.cs]: I've always been used to disposing things after any http requests, but according to [this](http://stackoverflow.com/questions/22560971/what-is-the-overhead-of-creating-a-new-httpclient-per-call-in-a-webapi-client) it seems it's meant to be reused.

File: RiotSharp/Requester.cs (hunk 1 above)
Comment (BenFradet) [inline RiotSharp/Requester.cs]: yup, I think it's better as `rootDomain`

File: RiotSharp/Requester.cs [hunk 6]
```diff
@@ -1,90 +1,124 @@
...
+using System.Net.Http;
 using System.Threading.Tasks;
 
 namespace RiotSharp
 {
     class Requester
     {
-        protected string rootDomain;
+        protected string RootDomain;
+        private readonly HttpClient _httpClient;
```
Comment (BenFradet) [inline RiotSharp/Requester.cs]: please use only camel case without the starting `_` to be coherent with the rest of the project, thanks

File: RiotSharpTest/App.config [hunk 7]
```diff
@@ -1,21 +1,21 @@
...
-        <add key="Team1Id" value="TEAM-66fbf960-a74c-11e4-a5a6-c81f66dd45c9"/>
-        <add key="Team2Id" value="TEAM-0a2ea1b0-f1cf-11e4-bc7e-c81f66ddabda"/>
-        <add key="GameId" value="1826202304"/>
-        <add key="ChampionId" value="38"/>
-        <add key="Region" value="na"/>
-        <add key="TournamentRegion" value="euw"/>
-        <add key="TournamentCode" value="EUW04197-bfc3c73a-6a50-4c9c-8c0b-044f222f9488"/>
-        <add key="MatchId" value="2440685384"/>
-        <add key="TournamentId" value="4272"/>
+      <add key="ApiKey" value="API_KEY"/>
```
Comment (BenFradet) [inline RiotSharpTest/App.config]: if you could leave the 4-indent, that'd be great

**Comment by BenFradet** [comment]
Tested locally and except for the few cosmetic remarks I made, I'm ready to merge.

**Comment by RejectKid** [comment]
alrighty pushed but the cosmetic changes.  btw do you have a visual studio style settings i could import so i dont keep stepping on your toes with the resharper style

**Comment by BenFradet** [comment]
Will merge tonight.
Otherwise, I use the default vs settings regarding formatting.

File: RiotSharpTest/App.config [hunk 8]
```diff
@@ -1,21 +1,21 @@
...
-        <add key="Team1Id" value="TEAM-66fbf960-a74c-11e4-a5a6-c81f66dd45c9"/>
-        <add key="Team2Id" value="TEAM-0a2ea1b0-f1cf-11e4-bc7e-c81f66ddabda"/>
-        <add key="GameId" value="1826202304"/>
-        <add key="ChampionId" value="38"/>
-        <add key="Region" value="na"/>
-        <add key="TournamentRegion" value="euw"/>
-        <add key="TournamentCode" value="EUW04197-bfc3c73a-6a50-4c9c-8c0b-044f222f9488"/>
-        <add key="MatchId" value="2440685384"/>
-        <add key="TournamentId" value="4272"/>
+    <add key="ApiKey" value="API_KEY"/>
```
Comment (BenFradet) [inline RiotSharpTest/App.config]: I meant 4 spaces more than `appSettings`, sorry

**Comment by BenFradet** [comment]
I'll merge and correct myself.

**Pull request merged**